test/e2e/upgrade_test.go

```diff
@@ -45,6 +45,12 @@ func upgradeTestSuite(t *testing.T) {
 	err = testCtx.configureUpgradeTest()
 	require.NoError(t, err, "error configuring upgrade")
 
+	// get current Windows node state
+	err = testCtx.waitForWindowsNodes(gc.numberOfMachineNodes, false, true, false)
+	require.NoError(t, err, "wrong number of Machine controller nodes found")
+	err = testCtx.waitForWindowsNodes(gc.numberOfBYOHNodes, false, true, true)
+	require.NoError(t, err, "wrong number of ConfigMap controller nodes found")
```

Comment:
Why are we waiting for the BYOH nodes here given BYOH nodes are not upgraded?

Reply:
This suite ends up calling `testPrometheus` which requires the BYOH nodes to loaded into the global node variable, which is done by waitForWindowsNodes as the tests currently stand.

I wanted to avoid this PR growing any more so I stuck with this pattern instead of introducing even more code and making things less consistent throughout the e2e's, I'm fine adding extra code to fix this if thats what you think is best, else it can be done within https://issues.redhat.com/browse/WINC-620

Reply:
No I am fine with it. I keep forgetting `waitForWindowsNodes` has that side effect.